middleware/common/include/common/flag.h

```diff
@@ -6,176 +6,168 @@
+      template< casual::common::flag::enum_flag_like T>
+      constexpr auto count( T flags)
+      {
+         using type = std::underlying_type_t< T>;
 
-         constexpr friend Flags operator + ( Flags lhs, Flags rhs) noexcept { return lhs | rhs;}
-         constexpr friend Flags& operator += ( Flags lhs, Flags rhs) noexcept { return lhs |= rhs;}
-         //! @}
+         if constexpr( std::is_unsigned_v< type>)
+            return std::popcount( std::to_underlying( flags));
+         else
+            return std::popcount( static_cast< std::make_unsigned_t< type>>( std::to_underlying( flags)));
+      }
 
+      //! @returns true if there are no bits set in `flag` outside the `expected` bits; 
+      template< casual::common::flag::enum_flag_like T>
+      constexpr bool valid( T expected, T flag)
+      {
+         return ( ~std::to_underlying( expected) & std::to_underlying( flag)) == 0;
+      }
 
-         constexpr friend std::ostream& operator << ( std::ostream& out, Flags flags)
+      //! @returns a converted flag type from `other` to the same type as `filter` that only has bits sets that
+      //! are set in `filter`
+      template< casual::common::flag::enum_flag_like T, typename U>
+      constexpr auto convert( T filter, U other) -> decltype( static_cast< T>( other))
+      {
+         return filter & static_cast< T>( other);
+      }
+   } // common::flag
+   
+   namespace common::stream::customization::supersede
+   {
+      template< common::flag::enum_flag_like T> 
+      struct point< T>
+      {  
+         //! tries to print the flag in the best representation possible
+         static void stream( std::ostream& out, T value)
          {
-            return print( out, flags, traits::priority::tag< 1>{});
-         }
+            // if we can use _description( enum)_ we use it.
+            if constexpr( common::flag::detail::has_description< T>)
+            {
+               out << '[';
 
-         CASUAL_FORWARD_SERIALIZE( m_flags)
+               using underlying_type = std::underlying_type_t< T>;
+               using unsigned_type = std::make_unsigned_t< underlying_type>;
+               constexpr auto bit_count = std::numeric_limits< unsigned_type>::digits;
+               std::bitset< bit_count> bits( std::to_underlying( value));
 
-      private:
+               algorithm::for_n< bit_count>( [ &out, bits]( unsigned_type index) mutable
+               {
+                  if( ! bits.test( index))
+                     return;
 
-         template< typename Enum>
-         static auto print( std::ostream& out, Flags< Enum> flags, traits::priority::tag< 1>) -> decltype( out << description( Enum{}))
-         {
-            // TODO maintainence: This can probably be done a lot better...
-            if( ! flags)
-               return out << "[]";
+                  // print the flag
+                  out << ' ' << description( static_cast< T>( unsigned_type( 1) << index));
 
```

Comment:
Noted that a number of the defined "Flag enums" ha a "no_flags" value defined (==0). Is it possible to be smart and output that description, if there is one? I do not think this code does that. Probaly prints "[]". Perfectly acceptable, but...

Reply:
Ahh, yeah, thats a thought... I don't think it's possible though since we can't know at compile-time if a given value has a _string representation_. We can't know if  `description( flag)` will return `no_flags` or `<unknown>` (which is our convention for absent string representation). Well, we can test for `<unknown>` but that seems a bit dodgy, to rely on a convention.

Reply:
But, we can do a compile-time opt in for _absent string representation_.
```cpp
enum struct Flag
{
   no_flags = 0,
   flag1 = 1,
   flag3 =  4,
};

consteval void casual_enum_as_flag( Flag);
consteval casual_enum_as_flag_has_empty( Flag);
```
Then we can chose to use `description( ...)` for empty flag compile-time for a give enum type.

But... A bit of a hassle for very little gain.